### PR TITLE
nim 0.16.0 + add linenoise option

### PR DESCRIPTION
**Pull request by subsetpark**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Add option to build the nim compiler with linenoise for readline-like
behavior in the REPL (nim secret).

### CODE REVIEW

**Comment by woodruffw** [comment]
@subsetpark Do you know if other distributions ship nim with this enabled? If so, we might want to enable it by default and do `build.without?` instead.

**Comment by subsetpark** [comment]
@woodruffw I don't believe so:
- It's not required on Windows;
- `koch boot`, the standard compiler bootstrap command, requires the `useLinenoise` flag to include it.

**Comment by ilovezfs** [comment]
https://github.com/gentoo/gentoo/blob/master/dev-lang/nim/nim-0.15.2.ebuild#L33

Would it be better to `depends_on "readline"`?

**Comment by woodruffw** [comment]
@subsetpark Got it, thanks for checking!

**Comment by subsetpark** [comment]
@woodruffw the standard tarball available on nim-lang.org comes with C source files, and a shell script to compile them (which the recipe runs); the functionality in this option is *not* available in those C files. Hence the recipe effectively has to generate new sources.

**Comment by subsetpark** [comment]
@ilovezfs `useGnuReadline` is actually an older flag that is now a synonym for `useLinenoise`; the project moved over to linenoise in order to drop that dependency.

**Comment by ilovezfs** [comment]
So the gentoo distro enables this by default.
https://github.com/gentoo/gentoo/blob/master/dev-lang/nim/nim-0.15.2.ebuild#L17

**Comment by subsetpark** [comment]
We definitely could too; it doesn't introduce any other dependencies. Basically on the pro side:
- You get to use readline commands in the repl

On the against side:
- It introduces two new compilation steps into the build;
- It's a pretty niche feature (The 'repl' is actually an undocumented command -- literally, `nim secret`; the actual nim compiler functions identically)

As a non-member I couldn't speak nearly well enough to the Homebrew 'philosophy' as to which side to favor.

**Comment by ilovezfs** [comment]
Since it doesn't add dependencies and it doesn't sound like anyone would have a reason to _want_ a build without it, it seems worth the extra build steps to avoid adding an option to the formula.

**Comment by subsetpark** [comment]
OK, I'll adjust.

 Today also has a new  release: 0.16.0. Shall I update the  versionin the same PR?

**Comment by woodruffw** [comment]
Yeah, adding 0.16.0 to this PR sounds good to me. No point in having two rebottles when we can make do with one :smile:

**Comment by ilovezfs** [comment]
>Make linenoise opt out

Let's just turn it on unconditionally without having an opt-out.

**Comment by subsetpark** [comment]
@woodruffw I added 0.16.0. That update deprecates a few things that the previous version was relying on, so there's some relatively substantial changes to the install process for both `head` and stable.

File: Formula/nim.rb [hunk 1]
```diff
@@ -63,6 +61,6 @@ def install
       license = "MIT"
       requires "nim >= 0.15.0"
     EOS
-    assert_equal "name: \"hello\"", shell_output("#{bin}/nimble dump").split("\n")[1].chomp
+    assert_equal "name: \"hello\"", shell_output("#{bin}/nimble dump").split("\n")[0].chomp
```
Comment (MikeMcQuaid) [inline Formula/nim.rb]: `shell_output("#{bin}/nimble dump").lines.first` would be preferable.

**Review by MikeMcQuaid — APPROVED** [review]

File: Formula/nim.rb (hunk 1 above)
Comment (nijikon) [inline Formula/nim.rb]: @subsetpark can you apply this please? I want to merge and then do #8647

Comment (subsetpark) [inline Formula/nim.rb]: @nijikon All done.

Comment (subsetpark) [inline Formula/nim.rb]: @nijikon This PR also takes care of the upgrade to 0.16.0 however.

**Comment by nijikon** [comment]
@subsetpark Oh, you updated to `0.16.0`, sweet 👍

**Review by nijikon — APPROVED** [review]

**Comment by nijikon** [comment]
You are awesome, thanks for your contribution! :tada:

**Pull request closed without merging**